core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -346,24 +347,26 @@ private EndpointGroup mapEndpoint(EndpointGroup endpointGroup) {
     private CompletableFuture<Boolean> initEndpoint(Endpoint endpoint) {
         endpointGroup = null;
         updateEndpoint(endpoint);
-        acquireEventLoop(endpoint);
-        return initFuture(true, null);
+        final boolean acquired = acquireEventLoop(endpoint);
+        return initFuture(acquired, null);
```

Comment:
Please check if whether it's okay to return `initFuture(false, null)` when exception is thrown in the `acquireEventLoop(endpoint)`

Reply:
Solved: https://github.com/line/armeria/pull/5489#discussion_r1545899786